openwisp_radius/api/permissions.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+from django.utils.translation import gettext_lazy as _
+from ipware import get_client_ip
+from rest_framework.exceptions import APIException, PermissionDenied
+from rest_framework.permissions import BasePermission
+
+logger = logging.getLogger(__name__)
+
+
+class IsSmsVerificationEnabled(BasePermission):
+    def has_permission(self, request, view):
+        organization = getattr(view, 'organization')
+        client_ip = get_client_ip(request)[0]
+        verification = self.has_sms_verification(view, organization)
+        if not verification:
+            logger.warning(
+                f'View {view.__class__.__name__} is being accessed for organization '
+                f'{organization.name} but SMS verification is disabled for '
+                f'this organization. Client IP address: {client_ip}'
+            )
+            raise PermissionDenied(
+                _('SMS verification is not enabled for this organization')
+            )
+        return verification
+
+    def has_sms_verification(self, view, org):
+        try:
+            verification = org.radius_settings.sms_verification
+        except Exception:
+            logger.exception(
+                f'Got exception while accessing radius_settings for {org.name}'
+            )
+            raise APIException(
+                _(
+                    'Could not complete operation '
+                    'because of an internal misconfiguration'
+                )
+            )
+        return verification
```

Comment:
please make this a reusable function in `openwisp_radius.api.utils` so we can eliminate the repetition between this code and `RegisterSerializer.is_sms_verification_enabled`.

eg:

```python
def is_sms_verification_enabled(organization):
    try:
        return organization.radius_settings.sms_verification
    except ObjectDoesNotExist:
        logger.exception(
            'Got exception while accessing '
            f'radius_settings for {organization.name}'
        )
        raise APIException(
            _(
                'Could not complete operation '
                'because of an internal misconfiguration'
            )
        )
```